src/System.Net.Http/src/System/Net/Http/Managed/HttpSystemProxy.cs

```diff
@@ -0,0 +1,162 @@
+            _sessionHandle = sessionHandle;
+
+            if (proxyHelper.ManualSettingsOnly)
+            {
+                _proxyUri = GetUriFromString(proxyHelper.Proxy);
+
+                if (!string.IsNullOrWhiteSpace(proxyHelper.ProxyBypass))
+                {
+                    // Process bypass list for manual setting.
+                    string[] list = proxyHelper.ProxyBypass.Split(';');
+                    List<string> tmpList = new List<string>();
+
+                    foreach (string value in list)
+                    {
+                        string tmp = value.Trim();
+                        if (tmp == "<local>")
+                        {
+                            _bypassLocal = true;
+                            continue;
+                        }
+                        if (tmp.Length > 0)
+                        {
+                            tmpList.Add(tmp);
+                        }
+                    }
+                    if (tmpList.Count > 0)
+                    {
+                        _bypass = tmpList.ToArray();
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// This function will evaluate given string and it will try to convert
+        /// it to Uri object. The string could contain URI fragment, IP address and  port
+        /// tuple or just IP address or name. It will return null if parsing fails.
+        /// </summary>
+        private static Uri GetUriFromString(string value)
+        {
+            if (string.IsNullOrEmpty(value))
+            {
+                return null;
+            }
+
+            if (!value.Contains("://"))
+            {
+                value = "http://" + value;
+            }
+
+            if (Uri.TryCreate(value, UriKind.Absolute, out Uri uri) &&
+                    (uri.Scheme == Uri.UriSchemeHttp || uri.Scheme == Uri.UriSchemeHttps))
+            {
+                // We only support http and https for now.
+                return uri;
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// Gets the proxy URI. (iWebProxy interface)
+        /// </summary>
+        public Uri GetProxy(Uri uri)
+        {
+            if (_proxyHelper.ManualSettingsOnly)
+            {
+                return _proxyUri;
+            }
+            var proxyInfo = new Interop.WinHttp.WINHTTP_PROXY_INFO();
+            if (_proxyHelper.GetProxyForUrl(_sessionHandle, uri, out proxyInfo))
+            {
+                return GetUriFromString(Marshal.PtrToStringUni(proxyInfo.Proxy));
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// Checks if URI is subject to proxy or not.
+        /// </summary>
+        public bool IsBypassed(Uri uri)
+        {
+            if (_proxyHelper.ManualSettingsOnly)
+            {
+                if (_bypassLocal)
+                {
+                    // TODO: implement bypass macth.
```

Comment:
Is this something you plan to do in this PR?  If not, please include the issue # in the TODO.

Reply:
>// TODO: implement bypass macth.

nit: spelling "macth" -> "match"

Reply:
I was not planning to - unless there are strong feelings about it. Can I refer original  #23150  @stephentoub or should I open new specific issue? I don't know exactly how the  bypass matching works on Windows so I need to do more exploration before coding.